### DISCUSSION TITLE
How to disable deploys for WIP/non-pr branches?

### DISCUSSION THREAD

**Discussion by kara-todd** [vercel/vercel#5642] [Help] [answered]
I tend to push branch work to the main GitHub repo fairly often in order to allow review of progress and for backup purposes. This means for branch urls I'm often committing work that I know is broken and I don't want builds to be run for. Is there a way to only allow builds for PRs? or only certain branches?

**Comment by styfle** [comment, marked as answer]
Hi @kara-todd 

Take a look at [How do I use the "Ignored Build Step" field on Vercel?](https://vercel.com/knowledge/how-do-i-use-the-ignored-build-step-field-on-vercel)

**Reply by kara-todd** [reply]
This worked for me. I used the config:

```
#!/bin/bash

if [[ "$BRANCH" == "qa" || "$BRANCH" == "master" ]] ; then
  # Proceed with the build
  exit 1;

else
  # Don't build
  exit 0;
fi
```

I had to add the value `$BRANCH` in the vercel console and mapped it to: `VERCEL_GIT_COMMIT_REF`. For some reason, using `$VERCEL_GIT_COMMIT_REF` directly did not return a value.
